students/JaeKim/session08/circle.py

```diff
@@ -0,0 +1,55 @@
+from math import pi
+
+class Circle:
+    def __init__ (self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        return pi * self.radius**2
+
+    def __str__(self):
+        return f'Circle with radius: {self.radius:.6f}'
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.radius})'
+
+    def __add__(self, other):
+        return f'{self.__class__.__name__}({self.radius + other.radius})'
+
+    def __mul__(self, other):
+        return f'{self.__class__.__name__}({self.radius * other.radius})'
+
+    def __gt__(self, other):
+        return {self.radius > other.radius}
+
+    def __eq__(self, other):
+        return {self.radius == other.radius}
+
+
+class Sphere(Circle):
+    def __str__(self):
+        return f'Sphere with radius: {self.radius:.6f}'
+
+    def __repr__(self):
```

Comment:
using __class__ here is really great -- why not also int the __str__ ?

Also, if you use __class__ here, then you can put it in the base class (Circle) and it will "just work" here!